core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -540,57 +571,76 @@ void processShareFetch(ShareFetchData shareFetchData) {
             return;
         }
 
-        try {
-            shareFetchData.partitionMaxBytes().keySet().forEach(topicIdPartition -> {
-                SharePartitionKey sharePartitionKey = sharePartitionKey(
-                    shareFetchData.groupId(),
-                    topicIdPartition
-                );
-                SharePartition sharePartition = getOrCreateSharePartition(sharePartitionKey);
+        // Initialize lazily, if required.
+        Map<TopicIdPartition, Throwable> erroneous = null;
+        Set<DelayedShareFetchKey> delayedShareFetchWatchKeys = new HashSet<>();
+        for (TopicIdPartition topicIdPartition : shareFetchData.partitionMaxBytes().keySet()) {
+            SharePartitionKey sharePartitionKey = sharePartitionKey(
+                shareFetchData.groupId(),
+                topicIdPartition
+            );
+
+            SharePartition sharePartition;
+            try {
+                sharePartition = getOrCreateSharePartition(sharePartitionKey);
+            } catch (Exception e) {
+                // Complete the whole fetch request with an exception if there is an error processing.
+                // The exception currently can be thrown only if there is an error while initializing
+                // the share partition. But skip the processing for other share partitions in the request
+                // as this situation is not expected.
+                log.error("Error processing share fetch request", e);
+                if (erroneous == null) {
+                    erroneous = new HashMap<>();
```

Comment:
It seems it's more intuitive to initialize erroneous as an empty map so that we don't need to deal with it being `null`.

Reply:
The exception occuring from `getOrCreateSharePartition` will be rare hence I was of an opinion of not initializing the erroneous and do that only when needed.